### DISCUSSION TITLE
Responsive Props

### DISCUSSION THREAD

**Discussion by kwameopareasiedu** [mantinedev/mantine#2830] [Feature requests]
Hello, I've been using Mantine for a while and it's sooo good. However, a feature that would make it even more awesome (I think), is **Responsive Props**.

I've read up using the `createStyled`, `useMediaQuery` and `<MediaQuery />` approaches on how to "style without using CSS". Except we do use CSS (E.g. `screen and (min-width: "xxxpx"`).

Also the media query approach becomes even more cumbersome if I have to use more than 2 breakpoint values (E.g. one for `xs`, one for `sm`, one for `md`, one for `lg`, one for `xl` and/or one for `>xl`). In this scenario, you'd end up with multiple `useMediaQuery`s or some cumbersome inline or `createStyles` specification **just for a single attribute of a single component**.

### **Create Styles** Approach
```tsx
// Responsive background color for xs, sm, md, lg and xl

const useStyles = createStyles((theme) => ({
  container: {
    backgroundColor: theme.colors.blue[6],

    [theme.fn.largerThan(theme.breakpoints.sm)]: {
      backgroundColor: theme.colors.pink[1],
    },

    [theme.fn.largerThan(theme.breakpoints.md)]: {
      backgroundColor: theme.colors.pink[2],
    },

    [theme.fn.largerThan(theme.breakpoints.lg)]: {
      backgroundColor: theme.colors.pink[3],
    },

    [theme.fn.largerThan(theme.breakpoints.xl)]: {
      backgroundColor: theme.colors.pink[4],
    },
  },
}));

const MyComponent = () => {
  const { classes } = useStyles();

  return <Box className={classes.container}>

    {/* Rest of app */}
  
  </Box>;
}
```

### **useMediaQuery** Approach
```tsx
// Responsive background color for xs, sm, md, lg and xl

const MyComponent = () => {
  const theme = useMantineTheme();
  const isSm = useMediaQuery(theme.fn.largerThan(theme.breakpoints.sm));
  const isMd = useMediaQuery(theme.fn.largerThan(theme.breakpoints.md));
  const isLg = useMediaQuery(theme.fn.largerThan(theme.breakpoints.lg));
  const isXl = useMediaQuery(theme.fn.largerThan(theme.breakpoints.xl));

  return <Box sx={{
      backgroundColor: isXl ? theme.colors.pink[4] : isLg ? theme.colors.pink[3] : isMd ? theme.colors.pink[2] : isSm ? theme.colors.pink[1] : theme.colors.pink[6]
    }}>

    {/* Rest of app */}

  </Box>;
}
```

### [ Proposed ] Responsive Props

```tsx
const MyComponent = () => {
  return <Box sx={{
      backgroundColor: [ 
        theme.colors.pink[4],
        theme.colors.pink[1],
        theme.colors.pink[2],
        theme.colors.pink[3],
        theme.colors.pink[4],
      ]
    }}>

    {/* Rest of app */}

  </Box>;
}
```

Is it possible to have responsive props in a future version of Mantine, or is it being worked on...??

**Comment by rtivital** [comment]
It is planned to introduce responsive component props like `mx`, `pt`, etc. It is not planned to modify `sx`/`createStyles` syntax this way.
